sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/FlexRulesDefinition.java

```diff
@@ -19,22 +19,31 @@
  */
 package org.sonar.plugins.flex;
 
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
 import org.sonar.api.server.rule.RulesDefinition;
 import org.sonar.flex.checks.CheckList;
 import org.sonar.plugins.flex.core.Flex;
-import org.sonar.squidbridge.annotations.AnnotationBasedRulesDefinition;
+import org.sonarsource.analyzer.commons.RuleMetadataLoader;
 
 public final class FlexRulesDefinition implements RulesDefinition {
 
   private static final String REPOSITORY_NAME = "SonarQube";
+  private static final String RESOURCE_BASE_PATH = "org/sonar/l10n/flex/rules/flex";
+  private static final Set<String> TEMPLATE_RULE_KEYS = new HashSet<>(Arrays.asList("XPath", "CommentRegularExpression"));
 
   @Override
   public void define(Context context) {
     NewRepository repository = context
       .createRepository(CheckList.REPOSITORY_KEY, Flex.KEY)
       .setName(REPOSITORY_NAME);
 
-    new AnnotationBasedRulesDefinition(repository, Flex.KEY).addRuleClasses(false, CheckList.getChecks());
+    // FIXME: with SonarQube 6.7, should use the sonar way profile location as extra parameter
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Take the required action to fix the issue indicated by this comment. ([squid:S1134](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.flex%3Aflex&branch=SONARFLEX-115&issues=AWGuSbKA-_vqYVDUCV73&open=AWGuSbKA-_vqYVDUCV73)

[ISSUE_KEY]: AWGuSbKA-_vqYVDUCV73